app/services/update_account_service.rb

```diff
@@ -0,0 +1,20 @@
+# frozen_string_literal: true
+
+class UpdateAccountService < BaseService
+  def call(account, params, raise_error: false)
+    was_locked = account.locked
+    update_method = raise_error ? :update! : :update
+    account.send(update_method, params).tap do |ret|
+      next unless ret
+      authorize_all_follow_requests(account) if was_locked && !account.locked
```

Comment:
Why not `authorize_all_follow_requests(account) if !ret && (was_locked && !account.locked)`?

Reply:
I want to explicitly state that "following codes are only for succeeded situation" .
As another reason, it makes easy to add other code like this.